modules/typesniffer/typesniffer_test.go

```diff
@@ -121,3 +122,15 @@ func TestDetectContentTypeFromReader(t *testing.T) {
 	assert.NoError(t, err)
 	assert.True(t, st.IsAudio())
 }
+
+func TestDetectContentTypeOgg(t *testing.T) {
+	oggAudio, _ := hex.DecodeString("4f67675300020000000000000000352f0000000000007dc39163011e01766f72626973000000000244ac0000000000000071020000000000b8014f6767530000")
+	st, err := DetectContentTypeFromReader(bytes.NewReader(oggAudio))
+	assert.NoError(t, err)
+	assert.True(t, st.IsAudio())
+
+	oggVideo, _ := hex.DecodeString("4f676753000200000000000000007d9747ef000000009b59daf3012a807468656f7261030201001e00110001e000010e00020000001e00000001000001000001")
```

Comment:
I think it is good enough for "guessing" because most file type detectors also do "guess"

From wiki:

![image](https://github.com/go-gitea/gitea/assets/2114189/f4d4ccff-7ad5-4aa1-abe8-0e0802cfdd89)

And xxd:

![image](https://github.com/go-gitea/gitea/assets/2114189/5ffc1305-8bc1-4a4a-a048-f5ec3cfdc88e)

Reply:
> Where is the `vorbis`/`theora` in there?

Hmm ... they were encoded as hex string, so the test case uses `hex.DecodeString` to decode the hex string to bytes first.